odbc/src/api/get_type_info_tests.rs

```diff
@@ -0,0 +1,182 @@
+use crate::{handles::definitions::*, SQLGetTypeInfo};
+use bson::Bson;
+use odbc_sys::{SqlDataType, SqlReturn};
+
+mod unit {
+    use super::*;
+    use mongo_odbc_core::Error;
+
+    fn validate_result_set(data_type: SqlDataType, expectations: Vec<&str>) {
+        let handle: *mut _ = &mut MongoHandle::Statement(Statement::with_state(
+            std::ptr::null_mut(),
+            StatementState::Allocated,
+        ));
+        unsafe {
+            let get_next = |stmt: &Statement| -> Result<bool, Error> {
+                stmt.mongo_statement
+                    .write()
+                    .unwrap()
+                    .as_mut()
+                    .unwrap()
+                    .next(None)
+            };
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetTypeInfo(handle as *mut _, data_type)
+            );
+            let stmt = (*handle).as_statement().unwrap();
+            // for each expectation, check that calling next succeeds and we get the expected value
+            expectations.iter().for_each(|name| {
+                let result = get_next(stmt);
+                assert!(result.unwrap());
+                assert_eq!(
+                    Bson::String((*name).to_string()),
+                    stmt.mongo_statement
+                        .write()
+                        .unwrap()
+                        .as_mut()
+                        .unwrap()
+                        .get_value(1)
+                        .unwrap()
+                        .unwrap()
+                )
+            });
+
+            // check there are no additional results in the set that were not expected
+            let result = get_next(stmt);
+            assert!(!result.unwrap());
+        }
+    }
+
+    #[test]
+    fn test_all_types() {
+        let expectations = vec![
+            "long",
+            "binData",
+            "array",
+            "bson",
+            "dbPointer",
+            "decimal",
+            "javascript",
+            "javascriptWithScope",
+            "maxKey",
+            "minKey",
+            "null",
+            "object",
+            "objectId",
+            "symbol",
+            "timestamp",
+            "undefined",
+            "int",
+            "bool",
+            "double",
+            "string",
+            "date",
+        ];
+        validate_result_set(SqlDataType(0), expectations);
```

Comment:
You'll need to use Smallint values directly instead after changing the signature for SQLGetTypeInfoW.

Reply:
👍 yes, I think I did so in most places and must have just missed a couple!